02-atmos/components/terraform/fetch-weather/main.tf

```diff
@@ -13,43 +13,32 @@ data "terraform_remote_state" "location" {
 }
 
 locals {
+  # Pulls Longitude and Latitude from the Remote State of fetch-location
   users_location_map = data.terraform_remote_state.location.outputs.users_location_map
+  lat                = local.users_location_map.lat
+  lon                = local.users_location_map.lon
 
-  lat = local.users_location_map.lat
-  lon = local.users_location_map.lon
+  # Curls Weather API for Location Data
+  location_url  = "https://api.weather.gov/points/${local.lat},${local.lon}"
+  location_data = jsondecode(data.http.fetch_location.body).properties
 
-  location_url = "https://www.metaweather.com/api/location/search/?lattlong=${local.lat},${local.lon}"
-
-  location_id      = local.location_data[0].woeid
-  weather_base_url = "https://www.metaweather.com/api/location"
-  weather_url      = var.date != null ? "${local.weather_base_url}/${local.location_id}/${var.date}" : "${local.weather_base_url}/${local.location_id}/"
-
-  location_data     = jsondecode(data.http.fetch_location.body)
-  full_weather_data = jsondecode(data.http.fetch_weather.body)
-
-  selected_weather_data = var.date != null ? local.full_weather_data[0] : local.full_weather_data.consolidated_weather[0]
+  # Curls Weather API for Forecast Data
+  weather_url  = (var.hourly_forecast) ? local.location_data.forecastHourly : local.location_data.forecast
```

Comment:
@milldr Just cause I'm interested, why the parentheses around the condition?

Reply:
personal preference. I think enclosing the logic before a ternary op makes it easier to follow, especially in more complex conditions